server/public/model/user.go

```diff
@@ -477,14 +475,24 @@ func (u *User) PreSave() {
 	}
 
 	if u.Password != "" {
-		u.Password = HashPassword(u.Password)
+		hashed, err := HashPassword(u.Password)
+		if errors.Is(err, bcrypt.ErrPasswordTooLong) {
+			return NewAppError("User.PreSave", "model.user.pre_save.password_too_long.app_error",
+				nil, "user_id="+u.Id, http.StatusBadRequest).Wrap(err)
```

Comment:
This feels very weird to have an HTTP error being throw in the model layer. I feel like it should only ever be thrown in the API layer. Or is this just some legacy thing we have to accept when using `AppError`?